drivers/led/lp50xx.c

```diff
@@ -276,18 +283,19 @@ static int lp50xx_init(const struct device *dev)
 	}
 
 	/* Check LED configuration found in DT */
-	if (config->num_leds > config->max_leds) {
-		LOG_ERR("%s: invalid number of LEDs %d (max %d)",
-			dev->name,
-			config->num_leds,
-			config->max_leds);
-		return -EINVAL;
-	}
 	for (led = 0; led < config->num_leds; led++) {
 		const struct led_info *led_info =
 			lp50xx_led_to_info(config, led);
 
-		if (led_info->num_colors > LP50XX_COLORS_PER_LED) {
+		used_chans += led_info->num_colors;
+
```

Comment:
This path assumes that the indexes (`index` property) found in DT are coherent. It should be checked.

Reply:
Let me see if I get it right. You mean to check if num_colors and the current index is valid on the current LED module? Do you mean to check the whole chain from index 0 to the highest configured value? Would you also check if the red LED (in a RGB module) matches the index value?

Reply:
Ok I added a check that verifies that and RGB LED is placed on a multiple of three index as stated in the datasheet.